### ISSUE TITLE
Better typings for https options

### ISSUE THREAD

**Issue by spcfran** [fastify/fastify#1089]
The typings for the `https` options: 

https://github.com/fastify/fastify/blob/22bf64c041fe9c067f0c847ccfbeeff614cf298f/fastify.d.ts#L98

only consist of `key`/`cert` of type `Buffer`. However the code in fastify using this option are just relaying this object to the `https.createServer`/`http2.createSecureServer` functions:

https://github.com/fastify/fastify/blob/22bf64c041fe9c067f0c847ccfbeeff614cf298f/fastify.js#L111-L113

Both of which accept more options as per node typings:

- [`https.createServer`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/efa449afccaf85ef2099425cee41d349b48ea0a1/types/node/index.d.ts#L1894)
- [`http2.createSecureServer`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/efa449afccaf85ef2099425cee41d349b48ea0a1/types/node/index.d.ts#L7883)

Both sets of options ultimately have as a common ancestor the [`tls.TlsOptions`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/efa449afccaf85ef2099425cee41d349b48ea0a1/types/node/index.d.ts#L5719) interface, which at the same time inherits from [`tls.SecureContextOptions`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/efa449afccaf85ef2099425cee41d349b48ea0a1/types/node/index.d.ts#L5719) where we can effectively see the `key`/`cert` props (which btw can have other types apart from `Buffer`) and some others, so perhaps a simple solution would be to augment the choices of `https` by just referencing the `TlsOptions` interface:
```typescript
import * as tls from 'tls';
// ...
interface ServerOptionsAsSecure extends ServerOptions {
  https: tls.TlsOptions
}
```
It's not a complete solution as it doesn't cover all the different options `http2` adds on top of those, but it would be step forward.

**Comment by mcollina** [comment]
Would you like to help out and send a PR?
Il giorno ven 24 ago 2018 alle 18:55 Fran Rodriguez <
notifications@github.com> ha scritto:

**Comment by spcfran** [comment]
With pleasure! I'll do it over the weekend

**Comment by mcollina** [comment]
Fixed in #1109
